apps/epl/src/epl_tracer.erl

```diff
@@ -82,6 +89,22 @@ init(Node) ->
                      {trace_ts, Pid, 'receive', Msg, _TS} ->
                          %% we count received messages and their sizes
                          Size = erts_debug:flat_size(Msg),
+                         case ets:lookup(epl_timeline_pids, Pid) of
+                             [] -> ok;
+                             _ ->
+                                 TimelineOld = case ets:lookup(epl_timeline_cache, Pid) of
+                                                   [] -> [];
+                                                   [{Pid, Timeline}]  -> Timeline
+
+                                               end,
+                                 NewValue = {Msg, sys:get_state(Pid)},
+                                 case Msg of
```

Comment:
Wouldn't it be a bit more performant if we'd first check whether the message is not a system message, and only then update the timeline?